pubsub-client/src/error.rs

```diff
@@ -0,0 +1,37 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum PubsubClientError {
```

Comment:
Can you please put the movement of this enum to its own module in a separate commit to make it easier to see the variant changes?